framework/CodeInterfaces/CobraTF/ctfdata.py

```diff
@@ -0,0 +1,551 @@
+                    startLineNumber, endLineNumber)}
+        return times
+
+    def getMajor(self, lines):
+        # Method that looks for key word MAJOR EDIT for reading major edit block
+        # @ In, lines, list of lines of the output file
+        # @ Out, majorDict, dict, dictionary containing the major edit info
+
+        # Booleans
+        # start output reading (skip 'input file listing' in the output)
+        out_time = False
+        out_time2 = False
+        o_type1 = False
+        o_type2 = False
+        o_type3 = False
+        o_type4 = False
+
+        # to check total number of channels & node
+        checkChannelNum = False
+        checkNodeNum = False
+
+        # count time step
+        timestep_cnt = 0
+        # number of averaged variables (total variables in major edit)
+        avg_variables_cnt = 15
+        # number of variables (total variables in major edit)
+        variables_cnt = 38
+        # number of channels
+        channel_cnt = 0
+        # number of axial nodes (channel)
+        node_cnt = 0
+        # number of fuel nodes
+        fuel_cnt = 0
+        count1 = 0
+        count2 = 0
+        # 1. count the total number of time steps, channels, axial nodes, and fuel rods.
+        for line in lines:
+            out_tag_avg = 'aver. properties for channels'
+            out_tag1 = 'fluid properties for channel'
+            out_tag4 = 'nuclear fuel rod no.'
+            # to check total number of channel
+            if 'subchannel data' in line:
+                checkChannelNum = True
+            if 'axial loss coefficients' in line:
+                checkChannelNum = False
+                checkNodeNum = True
+            if '------- *******************' in line:
+                checkNodeNum = False
+
+            if out_tag_avg in line:
+                # out_time : output reading start (stay 'True' once it is activated)
+                # out_time2: check sim_time evolution
+                out_time, out_time2 = True, True
+                timestep_cnt += 1
+
+            if out_tag1 in line:
+                o_type1 = True
+                chan_no = line.split()[9]
+                ##channel_cnt += 1
+
+            if out_tag4 in line:
+                o_type4 = True
+
+            # read total channel number
+            if (checkChannelNum == True) and (line != '\n') and (line.split()[0].isdigit()):
+                channel_cnt = int(line.split()[0])
+            # read total node number per channel
+            if (checkNodeNum == True) and (line != '\n') and (line.split()[0].isdigit()):
+                node_cnt = int(line.split()[len(line.split()) - 1])
+
+            # skip any blank line and lines that don't start with number
+            if (out_time == True) and (line != '\n') and (line.split()[0].isdigit()):
+                # filter few more lines that include characters
+                if (line.split()[1].replace('.', '', 1).isdigit()):
```

Comment:
can't you just use bool operations in fewer ```if```statements instead of a cascade of them?